### ISSUE TITLE
[REQUEST]: WebRTC for VisionOS

### ISSUE THREAD

**Issue by vodolazov** [Unity-Technologies/com.unity.webrtc#1019]
### Is your feature request related to a problem?

It'd be great to add support for VisionOS, hence VisionPro device.

**Comment by SetoKaiba** [comment]
I think the problem is that the webrtc from Google can't compile to vision os.

**Comment by karasusan** [comment]
@vodolazov 
I have never tested both VisionOS and Polyspacial.

**Comment by vodolazov** [comment]
Do you think it´d be possible to try with libwebrtc for ios? Supposedly ios and visionos are very close.
Anyways, thanks for looking at this.

**Comment by SetoKaiba** [comment]
> Do you think it´d be possible to try with libwebrtc for ios? Supposedly ios and visionos are very close. Anyways, thanks for looking at this.

No. The dynamic library has to be compiled for visionos directly.

**Comment by FejZa** [comment]
+1 for this request. @karasusan could you let us know, if this

1. maybe going to be looked at
2. definitely going to be looked at, just don't know when
3. definitely not going to be implemented

Thank you!

**Comment by AnandMWio** [comment]
+1 for this request.
anybody found a workaround or a solution?

**Comment by karasusan** [comment]
memo: WRS-507

**Comment by DavidSun0403** [comment]
Any upadtes for that till now?

**Comment by devluz** [comment]
I started porting [webrtc](https://github.com/because-why-not/webrtc_visionpro) and the [unity webrtc build](https://github.com/because-why-not/com.unity.webrtc_visionpro) to vision OS. If anyone wants to try this I uploaded a
[vision os test project](https://github.com/because-why-not/uwebrtc_visionpro_testproj.git)
Would be great if anyone can test it on the vision pro. So far I can only test in the simulator.

**Comment by AnandMWio** [comment]
> I started porting [webrtc](https://github.com/because-why-not/webrtc_visionpro) and the [unity webrtc build](https://github.com/because-why-not/com.unity.webrtc_visionpro) to vision OS. If anyone wants to try this I uploaded a [vision os test project](https://github.com/because-why-not/uwebrtc_visionpro_testproj.git) Would be great if anyone can test it on the vision pro. So far I can only test in the simulator.

Tested it on the Device, it works same as it works with the simulator. Great work! 
Webcam was not listed to select in one of the scene but, i think that is okay.

**Comment by devluz** [comment]
I have excluded the webcam because this needs more work. Great to know it works on the device already! I will build a proper release today. 

@karasusan do you think there is any good way to contribute this? I had to change the webrtc src folder and the src/build to get this to work.

**Comment by AnandMWio** [comment]
> I started porting [webrtc](https://github.com/because-why-not/webrtc_visionpro) and the [unity webrtc build](https://github.com/because-why-not/com.unity.webrtc_visionpro) to vision OS. If anyone wants to try this I uploaded a [vision os test project](https://github.com/because-why-not/uwebrtc_visionpro_testproj.git) Would be great if anyone can test it on the vision pro. So far I can only test in the simulator.

do you know why its not working for me when i copy it to my project to test it.

**Comment by devluz** [comment]
> > I started porting [webrtc](https://github.com/because-why-not/webrtc_visionpro) and the [unity webrtc build](https://github.com/because-why-not/com.unity.webrtc_visionpro) to vision OS. If anyone wants to try this I uploaded a [vision os test project](https://github.com/because-why-not/uwebrtc_visionpro_testproj.git) Would be great if anyone can test it on the vision pro. So far I can only test in the simulator.
> 
> do you know why its not working for me when i copy it to my project to test it.

You can create an issue at the test project repository then I look into it. Either the build log or the console log in xcode should have some clues.

**Comment by AnandMWio** [comment]
> > > I started porting [webrtc](https://github.com/because-why-not/webrtc_visionpro) and the [unity webrtc build](https://github.com/because-why-not/com.unity.webrtc_visionpro) to vision OS. If anyone wants to try this I uploaded a [vision os test project](https://github.com/because-why-not/uwebrtc_visionpro_testproj.git) Would be great if anyone can test it on the vision pro. So far I can only test in the simulator.
> > 
> > 
> > do you know why its not working for me when i copy it to my project to test it.
> 
> You can create an issue at the test project repository then I look into it. Either the build log or the console log in xcode should have some clues.

Thanks, i've created the issue - https://github.com/because-why-not/uwebrtc_visionpro_testproj/issues/2#issue-2213337740
